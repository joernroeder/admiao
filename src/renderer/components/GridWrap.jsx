import styled from '@emotion/styled'
import { Flex } from 'reflexbox'
import { height, minHeight } from 'styled-system'

const GridWrap = styled(Flex)`
	${height};
	${minHeight};
`

GridWrap.defaultProps = {
	mx: -1,
	backgroundColor: 'white',
	color: 'black',
}

export default GridWrap
