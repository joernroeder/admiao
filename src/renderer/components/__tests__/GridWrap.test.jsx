import React from 'react'

import { render } from '@testing-library/react'

import GridWrap from '../GridWrap'

describe('GridWrap component', () => {
	it('should correctly even out the outer cell gutter', () => {
		const { container } = render(<GridWrap />)
		const wrap = container.firstChild

		const margin = '-4px'

		expect(wrap).toHaveStyleRule('margin-left', margin)
		expect(wrap).toHaveStyleRule('margin-right', margin)

		expect(wrap).toMatchSnapshot()
	})
})
