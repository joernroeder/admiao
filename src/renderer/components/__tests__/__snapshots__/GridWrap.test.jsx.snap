// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridWrap component should correctly even out the outer cell gutter 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-left: -4px;
  margin-right: -4px;
  background-color: white;
  color: black;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  class="emotion-0"
/>
`;
